controllers/src/main/java/com/redhat/ipaas/controllers/IntegrationController.java

```diff
@@ -0,0 +1,157 @@
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * This class tracks changes to Integrations and attempts to process them so that
+ * their current status matches their desired status.
+ */
+@Service
+public class IntegrationController {
+
+    private final DataManager dataManager;
+    private final EventBus eventBus;
+    private final HashMap<Integration.Status, WorkflowHandler> handlers = new HashMap<>();
+    private ExecutorService executor;
+    private ScheduledExecutorService scheduler;
+    private long retrySeconds = 60;
+
+    @Autowired
+    public IntegrationController(DataManager dataManager, EventBus eventBus, List<WorkflowHandler> handlers) {
+        this.dataManager = dataManager;
+        this.eventBus = eventBus;
+        for (WorkflowHandler handler : handlers) {
+            for (Integration.Status status : handler.getTriggerStatuses()) {
+                this.handlers.put(status, handler);
+            }
+        }
+    }
+
+    @PostConstruct
+    public void start() {
+        executor = Executors.newSingleThreadExecutor();
+        scheduler = Executors.newScheduledThreadPool(1);
+        scanIntegrationsForWork();
+
+        eventBus.subscribe("integration-controller", (event, data) -> {
+            if ("change-event".equals(event)) {
+                try {
+                    ChangeEvent changeEvent = Json.mapper().readValue(data, ChangeEvent.class);
+                    if (changeEvent != null || changeEvent.getKind().isPresent() && changeEvent.getId().isPresent()) {
+                        if (Kind.from(changeEvent.getKind().get()) == Kind.Integration) {
+                            enqueueCheckIntegration(changeEvent.getId().get());
+                        }
+                    }
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+            }
+        });
+    }
+
+    @PreDestroy
+    public void stop() {
+        eventBus.unsubscribe("integration-controller");
+        scheduler.shutdownNow();
+        executor.shutdownNow();
+    }
+
+    private void scanIntegrationsForWork() {
+        dataManager.fetchAll(Integration.class).getItems().forEach(integration -> {
+            checkIntegration(integration);
+        });
+    }
+
+    private void enqueueCheckIntegration(String id) {
+        executor.execute(() -> {
+            checkIntegration(dataManager.fetch(Integration.class, id));
+        });
+    }
+
+    private void checkIntegration(Integration integration) {
+        Integration.Status desired = integration.getDesiredStatus().orElse(Integration.Status.Draft);
+        Integration.Status current = integration.getCurrentStatus().orElse(Integration.Status.Draft);
+        if (current != desired) {
```

Comment:
want to pull in/rebase and add the deployment data from https://github.com/redhat-ipaas/ipaas-rest/pull/231?